File: src/MaksIT.CertsUI/Models/APIKeys/Search/SearchAPIKeyRequest.cs
using System.Linq.Expressions;
using MaksIT.Core.Webapi.Models;

namespace MaksIT.CertsUI.Models.APIKeys.Search;

public class SearchAPIKeyRequest : PagedRequest {

  public string? OrganizationFilters { get; set; }

  public string? ApplicationFilters { get; set; }

  public Expression<Func<T, bool>> BuildOrganizationFilterExpression<T>() {
    return BuildFilterExpression<T>(OrganizationFilters);
  }

  public Expression<Func<T, bool>> BuildApplicationFilterExpression<T>() {
    return BuildFilterExpression<T>(ApplicationFilters);
  }
}

File: src/MaksIT.WebUI/src/models/certsUI/apiKeys/search/SearchAPIKeyRequest.ts
import { PagedRequest } from '../../../PagedRequest'

/** Mirrors <c>MaksIT.CertsUI.Models.APIKeys.Search.SearchAPIKeyRequest</c>. */
export interface SearchAPIKeyRequest extends PagedRequest {
  organizationFilters?: string | null
  applicationFilters?: string | null
}
